llvm/lib/Target/RISCV/RISCVRegisterInfo.cpp

```diff
@@ -511,6 +511,15 @@ bool RISCVRegisterInfo::eliminateFrameIndex(MachineBasicBlock::iterator II,
       // We can encode an add with 12 bit signed immediate in the immediate
       // operand of our user instruction.  As a result, the remaining
       // offset can by construction, at worst, a LUI and a ADD.
+
+      // Special case, try to split the offset across two ADDIs.
+      uint64_t ValSubLo12 = (uint64_t)Val - (uint64_t)Lo12;
+      const RISCVFrameLowering *TFI = getFrameLowering(MF);
+      const uint64_t Align = TFI->getStackAlign().value();
```

Comment:
I'm not sure why StackAlign matters here. You're not changing the stack pointer.